functorch/experimental/cond.py

```diff
@@ -115,6 +123,17 @@ def inner(pred, true_fn, false_fn, operands):
     return res
 
 
+@cond.py_impl(FakeTensorMode)
+def cond_fake_tensor_mode(pred, true_fn, false_fn, operands):
+    return true_fn(*operands)
+
+
+@cond.py_impl(DispatchKey.PythonDispatcher)
+def cond_python_dispatcher(*args):
+    _ = ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.PythonDispatcher))
+    return cond(*args)
```

Comment:
nit: this is just cond.fallthrough(DispatchKey.PythonDispatcher), ditto for the map_python_dispatcher

Reply:
@zou3519 I tried to use `fallthrough()` directly, but seems like the current behavior of `fallthrough()` will always add `PythonDispatcher` key back which causes infinite loop: 
```
    def _fallthrough_fn(self, operator, dispatch_key):
        def inner(*args, **kwargs):
            all_keys_after_current = torch._C._dispatch_keyset_full_after(dispatch_key)
            ...
```
and 
```
  constexpr DispatchKeySet(FullAfter, DispatchKey t)
      // LSB after t are OK, but not t itself.
      // "functionalities" have a notion of ordering (e.g. Autograd > Sparse >
      // Quantized > Dense). But backends don't really have an ordering.
      // Therefore, we're enforcing that FullAfter can only be used on
      // "functionality" keys.
      : repr_(
            (1ULL
             << (num_backends + static_cast<uint8_t>(toFunctionalityKey(t)) -
                 1)) -
            1) {
    *this = add(DispatchKey::PythonDispatcher);
  }
```
not sure what's the optimal thing to do here other than just masking the PythonDispatcher key and go.

Reply:
can you file an issue about this. iirc there is a problem with fallthrough interaction and python dispatcher

Reply:
@ezyang filed: https://github.com/pytorch/pytorch/issues/89037

Reply:
If you want to get this PR shipped to try stuff, no need to wait until that issue is fixed though we should figure it out at some point

Reply:
@zou3519 understood, I'll make a comment here pointing to the issue number.